doc/source/contributing.rst

```diff
@@ -4,13 +4,574 @@
+  in the Scientific Python community. This standard specifies the format of
+  the different sections of the docstring. See `this document
+  <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_
+  for a detailed explanation, or look at some of the existing functions to
+  extend it in a similar manner.
+
+- The tutorials make heavy use of the `ipython directive
+  <http://matplotlib.org/sampledoc/ipython_directive.html>`_ sphinx extension.
+  This directive lets you put code in the documentation which will be run
+  during the doc build. For example:
+
+  ::
+
+      .. ipython:: python
+
+          x = 2
+          x**3
+
+  will be rendered as
+
+  ::
+
+      In [1]: x = 2
+
+      In [2]: x**3
+      Out[2]: 8
+
+  This means that almost all code examples in the docs are always run (and the
+  output saved) during the doc build. This way, they will always be up to date,
+  but it makes the doc building a bit more complex.
+
+
+How to build the pandas documentation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requirements
+""""""""""""
+
+To build the *pandas* docs there are some extra requirements: you will need to
+have ``sphinx`` and ``ipython`` installed. `numpydoc
+<https://github.com/numpy/numpydoc>`_ is used to parse the docstrings that
+follow the Numpy Docstring Standard (see above), but you don't need to install
+this because a local copy of ``numpydoc`` is included in the *pandas* source
+code.
+
+Furthermore, it is recommended to have all `optional dependencies
+<http://pandas.pydata.org/pandas-docs/dev/install.html#optional-dependencies>`_
+installed. This is not needed, but be aware that you will see some error
+messages. Because all the code in the documentation is executed during the doc
+build, the examples using this optional dependencies will generate errors.
+Run ``pd.show_versions()`` to get an overview of the installed version of all
+dependencies.
+
+.. warning::
+
+   Sphinx version >= 1.2.2 or the older 1.1.3 is required.
+
+Building the documentation
+""""""""""""""""""""""""""
+
+So how do you build the docs? Navigate to your local  the folder
+``pandas/doc/`` directory in the console and run::
+
+    python make.py html
+
+And then you can find the html output in the folder ``pandas/doc/build/html/``.
+
+The first time it will take quite a while, because it has to run all the code
+examples in the documentation and build all generated docstring pages.
+In subsequent evocations, sphinx will try to only build the pages that have
+been modified.
+
+If you want to do a full clean build, do::
+
+    python make.py clean
+    python make.py build
+
+
+Starting with 0.13.1 you can tell ``make.py`` to compile only a single section
+of the docs, greatly reducing the turn-around time for checking your changes.
+You will be prompted to delete `.rst` files that aren't required, since the
+last committed version can always be restored from git.
+
```

Comment:
It is indeed correct, I only think the phrasing can be interpreted a bit wrongly -> it is *not a problem to delete* the files because git will remember them (you just should not commit the deletions, only the changes).

The problem is that if the files are not deleted, you get a warnings about files that are not referenced. This is not really a problem, but that way the real warnings about the thing you are writing are less visible I think. But maybe there is a way to tell sphinx to ignore them.